### PR TITLE
Nifi 2381

### PR DESCRIPTION
**Pull request by ToivoAdams**
Connection Pooling Service -Drop invalid connections and create new ones done.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-api/src/main/java/org/apache/nifi/dbcp/DBCPService.java [hunk 1]
```diff
@@ -31,4 +31,5 @@
 @CapabilityDescription("Provides Database Connection Pooling Service. Connections can be asked from pool and returned after usage.")
 public interface DBCPService extends ControllerService {
     public Connection getConnection()  throws ProcessException;
+
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-api/src/main/java/org/apache/nifi/dbcp/DBCPService.java]: Unnecessary change

File: nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/pom.xml [hunk 2]
```diff
@@ -52,7 +52,32 @@
...
+		    <scope>test</scope>
+		</dependency>
+		<dependency>
+    		<groupId>com.h2database</groupId>
+    		<artifactId>h2</artifactId>
+    		<version>1.4.192</version>
+    		<scope>test</scope>
+		</dependency>
+		
+		<dependency>
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/pom.xml]: Can you take care of indentation?

File: nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java [hunk 3]
```diff
@@ -114,6 +114,17 @@
         .sensitive(false)
         .build();
 
+    public static final PropertyDescriptor VALIDATION_QUERY = new PropertyDescriptor.Builder()
+            .name("Validation query")
+            .description("Validation query used to validate connections before returning them."
+                + "When connection is invalid, it get's dropped and new valid connection will be returned."
+                + "Note!! Using validation might have some performance penalty.")
+            .defaultValue(null)
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java]: I don't think default value to null is necessary.

File: nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java [hunk 4]
```diff
@@ -114,6 +114,17 @@
         .sensitive(false)
         .build();
 
+    public static final PropertyDescriptor VALIDATION_QUERY = new PropertyDescriptor.Builder()
+            .name("Validation query")
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java]: Please when adding new property, think about .displayName() and .name()

File: nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java [hunk 5]
```diff
@@ -114,6 +114,17 @@
...
+            .description("Validation query used to validate connections before returning them."
+                + "When connection is invalid, it get's dropped and new valid connection will be returned."
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java]: Cosmetic: white spaces in the description ;)

File: nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java [hunk 6]
```diff
@@ -171,6 +184,11 @@ public void onConfigured(final ConfigurationContext context) throws Initializati
         dataSource.setMaxWait(maxWaitMillis);
         dataSource.setMaxActive(maxTotal);
 
+        if (validationQuery!=null && validationQuery.isEmpty()==false) {
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java]:
````
validationQuery.isEmpty()==false
````
to be replaced by
````
!validationQuery.isEmpty()
````

**Comment by ToivoAdams** [comment]
Hi @pvillard31 

I did changes.
Thank you for reviewing.
Toivo
